istioctl/pkg/install/verify.go

```diff
@@ -31,39 +37,154 @@ import (
 	"k8s.io/client-go/kubernetes/scheme"
 
 	"istio.io/istio/galley/pkg/config/meta/metadata"
-	"istio.io/istio/pilot/pkg/serviceregistry/kube/controller"
 )
 
 var (
 	verifyInstallCmd *cobra.Command
 )
 
-func verifyInstall(enableVerbose bool, istioNamespaceFlag string,
-	restClientGetter genericclioptions.RESTClientGetter, options resource.FilenameOptions,
-	writer io.Writer, args []string) error {
-	if len(options.Filenames) == 0 {
-		if len(args) != 0 {
-			_, _ = fmt.Fprint(writer, verifyInstallCmd.UsageString())
-			return fmt.Errorf("verify-install takes no arguments to perform installation pre-check")
+type verifyInstallArgs struct {
+	kubeConfigFlags *genericclioptions.ConfigFlags
+	fileNameFlags   *genericclioptions.FileNameFlags
+	istioNamespace  string
+	set             []string
+	enableVerbose   bool
+	preCheck        bool
+}
+
+// NewVerifyCommand creates a new command for verifying Istio Installation Status
+func NewVerifyCommand() *cobra.Command {
+	vArgs := &verifyInstallArgs{}
+	verifyInstallCmd = &cobra.Command{
+		Use:   "verify-install",
+		Short: "Verifies Istio Installation Status or performs pre-check for the cluster before Istio installation",
+		Long: `
+		verify-install verifies Istio installation status against the installation file or the IstioControlPlane 
+		CustomResource you specified when you installed Istio. It loops through all the installation
+		resources defined in your installation file and reports whether all of them are
+		in ready status. It will report failure when any of them are not ready.
+
+		If you do not specify installation file or specify --pre-check it will perform pre-check for your cluster
+		and report whether the cluster is ready for Istio installation.
+`,
+		Example: `
+		# Verify that Istio can be freshly installed
```

Comment:
why doesn't this just check to see if my currently installed cluster is healthy?

Reply:
verify-install has two functions from the beginning: 
1. do the install pre-check to make sure the cluster is suitable for Istio installation
2. do post-check to make sure Istio is installed successfully

So we need keep these two functions there with new function supported.

Reply:
It seems like I would want a single command that tells me everything is ready to install. Then once I install, it tells me everything is healthy. Having to pass parameters is a worse UX than existing tools like linkerd and helm

Reply:
Agreed with @howardjohn here, though maybe unrelated to this issue/PR? Maybe we open another issue?

Reply:
I strongly agree.  The current `verify-install` is difficult to use because it requires the user to have the .yaml that was used to install Istio.

I propose in https://github.com/istio/istio/issues/20000 that the Istio install preserves enough stuff to allow `verify-install` to work without parameters.  I will work with Iris to create a proposal and get it implemented.

Reply:
In this PR, we support both IstioControlPlane cr as input: ```eg: verify-install -f example-istiocontrolplane.cr.yaml```

we also support ``` --set as input, eg: verify-install -s profile=minimal``` .